### ISSUE TITLE
cannot converge use tf.nn.l2_normalize

### ISSUE THREAD

**Issue by stevechoris** [williamleif/GraphSAGE#77]
I try graphsage with my own code and data.
it works well when not use tf.nn.l2_normalize after encoder and before decoder. but when i use tf.nn.l2_normalize to normlize the output vector to length of 1, the model just cannot converge and the loss does not decrease. what is the cause of this problem.

**Comment by RexYing** [comment]
I think this might be an specific issue about your dataset. In my experiment however, I have not encountered this before. With l2_normalize, the model essentially embeds the nodes into a high-dimensional sphere. It might not be good for some link prediction / reconstruction task where the underlying graph is not suitable to be embedded into sphere. Examples are tree-like, hierarchical graphs.
